docs/source/app_developers_guide/txn_submit_tutorial/_segments/header_info.rst

```diff
@@ -0,0 +1 @@
+Remember from :doc:`/architecture/transactions_and_batches`, that *inputs* and *outputs* are state addresses that the Transaction is allowed to read from or write to. In this case we used a six character prefix rather than full 70-character addresses, allowing us to work with any address in the IntKey subtree. Also remember that *dependencies* are the *header signatures* of Transactions that must be committed before ours (none in this case).
```

Comment:
Not sure I would phrase this reference this way.  It implies that the app developer guide reader is the same audience as the architecture doc reader.